api/src/test/java/org/apache/iceberg/types/TestSerializableTypes.java

```diff
@@ -63,7 +63,14 @@ public void testEqualTypes() throws Exception {
           Types.DecimalType.of(9, 3),
           Types.DecimalType.of(11, 0),
           Types.FixedType.ofLength(4),
-          Types.FixedType.ofLength(34)
+          Types.FixedType.ofLength(34),
+          Types.GeometryType.get(),
+          Types.GeometryType.of("srid:3857"),
+          Types.GeometryType.of("projjson:Test_Identifier"),
```

Comment:
Is this valuable? These tests are the same for any string.

Reply:
PrimitiveTypes are serialized and deserialized using `toString` and `fromTypeString`, since PrimitiveType has implemented `Object writeReplace()` method to handle Java serialization. Type parameters do affect how primitive types are serialized and deserialized.

A better place for such tests could be `TestTypes.java`, but I think that having tests here to ensure round-trip type consistency also makes sense.

Reply:
Sorry, I meant is it valuable to have a `srid` case and a `projjson` case. Both are strings and if we validate that any string is preserved by serialization and deserialization, I think that is sufficient.

Reply:
I'm removing the case with projjson and only preserves the one with srid.